apiserver/facades/client/action/run.go

```diff
@@ -58,103 +53,50 @@ func (a *ActionAPI) RunOnAllMachines(ctx context.Context, run params.RunParams)
 	if err != nil {
-		return results, errors.Trace(err)
+		return results, errors.Capture(err)
 	}
 
 	if modelInfo.Type != model.IAAS {
 		return results, errors.Errorf("cannot run on all machines with a %s model", modelInfo.Type)
 	}
 
-	return results, errors.NotSupportedf("actions in Dqlite")
+	args, err := newOperationTaskParams(run.Commands, run.Timeout, run.Parallel, run.ExecutionGroup)
+	if err != nil {
+		return results, errors.Capture(err)
+	}
+
+	result, err := a.operationService.RunOnAllMachines(ctx, args)
+	if err != nil {
+		return results, errors.Capture(err)
+	}
+
+	return toEnqueuedActions(result), errors.Capture(err)
 }
 
-func (a *ActionAPI) createRunActionsParams(
-	actionReceiverTags []names.Tag,
+func newOperationTaskParams(
 	quotedCommands string,
 	timeout time.Duration,
 	parallel *bool,
 	executionGroup *string,
-) (params.Actions, error) {
-	apiActionParams := params.Actions{Actions: []params.Action{}}
-
+) (operation.TaskParams, error) {
 	if actions.HasJujuExecAction(quotedCommands) {
-		return apiActionParams, errors.NewNotSupported(nil, fmt.Sprintf("cannot use %q as an action command", quotedCommands))
+		return operation.TaskParams{}, jujuerrors.NewNotSupported(nil, fmt.Sprintf("cannot use %q as an action command",
+			quotedCommands))
 	}
 
 	actionParams := map[string]interface{}{}
 	actionParams["command"] = quotedCommands
 	actionParams["timeout"] = timeout.Nanoseconds()
 
-	for _, tag := range actionReceiverTags {
-		apiActionParams.Actions = append(apiActionParams.Actions, params.Action{
-			Receiver:       tag.String(),
-			Name:           actions.JujuExecActionName,
-			Parameters:     actionParams,
-			Parallel:       parallel,
-			ExecutionGroup: executionGroup,
-		})
-	}
-
-	return apiActionParams, nil
-}
-
-// getAllUnitNames returns a sequence of valid Unit objects from state. If any
-// of the application names or unit names are not found, an error is returned.
-func (a *ActionAPI) getAllUnitNames(ctx context.Context, units, applications []string) (result []names.Tag, err error) {
-	var leaders map[string]string
-	getLeader := func(appName string) (string, error) {
-		if leaders == nil {
-			var err error
-			leaders, err = a.leadership.Leaders()
-			if err != nil {
-				return "", err
-			}
-		}
-		if leader, ok := leaders[appName]; ok {
-			return leader, nil
-		}
-		return "", errors.Errorf("could not determine leader for %q", appName)
-	}
-
-	// Replace units matching $app/leader with the appropriate unit for
-	// the leader.
-	unitsSet := set.NewStrings()
-	for _, unit := range units {
-		if !strings.HasSuffix(unit, "leader") {
-			unitsSet.Add(unit)
-			continue
-		}
-
-		app := strings.Split(unit, "/")[0]
-		leaderUnit, err := getLeader(app)
-		if err != nil {
-			return nil, apiservererrors.ServerError(err)
-		}
-
-		unitsSet.Add(leaderUnit)
-	}
-
-	for _, aName := range applications {
-		unitNames, err := a.applicationService.GetUnitNamesForApplication(ctx, aName)
```

Comment:
todo: remove no longer used methods from the applicationService provided will not be used by @nvinuesa PR.